modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/src/Controller/DaxkoLinkController.php

```diff
@@ -121,24 +122,32 @@ public function backlink(Request $request) {
     $userDetails = $this->daxkoClient->getRequest('members/' . $userData->member_id);
     // Check if this user is an active client.
     if ($userDetails->active) {
-      return new JsonResponse(
-        [
-          'error' => 0,
-          'token' => $token,
-          'user' => [
-            'name' => $userDetails->name->first_name,
-            'email' => $userDetails->emails[0]->email,
-          ],
-        ]
-      );
+      // Create drupal user if it doesn't exist and login it.
+      $name = $userDetails->name->first_name . ' ' . $userDetails->name->last_name;
+      $account = user_load_by_name($name);
```

Comment:
I think we have to load by email.
It is more unique. There could be 2 similar persons and this logic will break